frontend/src/Views/MainPage/MainPage.tsx

```diff
@@ -97,38 +97,61 @@ const MainPage: React.FC = () => {
    *            ** In the future, also update the ProgressBar status **
    * @param     formMovement: string => A movement that indicates if you go back or forward
    */
-  const handleFormState = (cmd?: string) => {
-    let moduleID = formStates[currForm].id;
-    let secondModuleID = null;
+  const handleFormState = (cmd: string, toggle?: boolean, status?: string) => {
+    // currModuleID can be any InputModule that's passed to this handler
+    let currModuleID = null;
+    // nextModuleID is the last activated InputModule, they are activated through the Next button
+    // which uses 'cmd = forward' argument
+    let nextModuleID = null;
+    // Respect ESLint no-param reassign
+    let tgl = toggle;
 
     switch (cmd) {
       case 'back':
         if (currForm >= 0) {
-          secondModuleID = formStates[currForm - 1].id;
+          currModuleID = formStates[currForm].id;
+          nextModuleID = formStates[currForm - 1].id;
           setCurrForm((prevForm) => prevForm - 1);
+          tgl = true;
         }
         break;
       case 'forward':
         if (formStates[currForm + 1]) {
-          secondModuleID = formStates[currForm + 1].id;
+          currModuleID = formStates[currForm].id;
+          nextModuleID = formStates[currForm + 1].id;
           setCurrForm((prevForm) => prevForm + 1);
+          tgl = true;
         }
         break;
       default:
-        if (cmd && Object.keys(InputModules).includes(cmd)) {
-          moduleID = cmd;
-        }
+        currModuleID = cmd;
         break;
     }
+
     const updatedStates = formStates.map((module: InputModuleInterface) => {
-      if (module.id === moduleID || module.id === secondModuleID) {
-        return {
-          ...module,
-          enable: !module.enable,
-        };
+      const newState = {
+        ...module,
+      };
+      if (module.id === currModuleID) {
+        newState.enable = tgl ? !newState.enable : newState.enable;
+        if (
+          (newState.status === 'active' || newState.status === 'warning')
```

Comment:
Not sure how others feel about it, but one way I like to simplify these is to do something like 
```ts
['active', 'warning'].includes(newState.status)
```
It's a little easier to read, but its a little less performant. Either way works though

Reply:
I'll try this out. I was so bothered by this and spent some time trying to think of a better way to write it instead of actually doing. 

I guess just doing it and having folk take a look was the best way in the end :smile:
Thank you for the tip!